### PR TITLE
Category-permission mapping extracted from AuthenticationProvider

### PR DESCRIPTION
**Pull request by trombka**
Solves https://github.com/TouK/nussknacker/blob/staging/engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/AuthenticationProvider.scala#L15
AuthenticationProvider no longer delivers a LoggedUser but an AuthenticatedUser instance with a roles collection. Permission mapping is moved to an additional LoggedUser apply method.

### CODE REVIEW

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala [hunk 1]
```diff
@@ -11,19 +15,28 @@ sealed trait LoggedUser {
...
-             categoryPermissions: Map[String, Set[Permission]] = Map.empty,
-             globalPermissions: List[GlobalPermission] = Nil,
-             isAdmin: Boolean = false): LoggedUser = {
-     if (isAdmin) {
-       AdminUser(id, username)
-     } else {
-       CommonUser(id, username, categoryPermissions, globalPermissions)
-     }
+  def apply(authenticatedUser: AuthenticatedUser, config: Config, processCategories: List[String]): LoggedUser = {
+    apply(authenticatedUser, AuthenticationConfiguration.getRules(config), processCategories)
```
Comment (mproch) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala:19]: we read config with each invocation here. Could we do this only once at application start?

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala [hunk 2]
```diff
@@ -11,19 +15,28 @@ sealed trait LoggedUser {
...
+            isAdmin: Boolean = false): LoggedUser = {
+    if (isAdmin) {
+      AdminUser(id, username)
+    } else {
+      CommonUser(id, username, categoryPermissions, globalPermissions)
+    }
   }
 
-  def apply(id: String, username: String, rulesSet: RulesSet): LoggedUser = {
+  def apply(id: String, username: String, roles: List[String], rulesSet: RulesSet): LoggedUser = {
```
Comment (mproch) [inline engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala:39]: roles are not used here?

File: engine/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/ExampleOAuth2ServiceFactorySpec.scala [hunk 3]
```diff
@@ -65,18 +65,22 @@ class ExampleOAuth2ServiceFactorySpec extends FlatSpec with Matchers with Patien
     }.futureValue
   }
 
+/* These test probably test the behaviour that is not desired any more
+i.e. assigning category permissions directly by an AuthenticationProvider
+and not by a roles-to-category-permission mapping
```
[the remark this answers is not in the document]
Comment (trombka) [reply engine/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/ExampleOAuth2ServiceFactorySpec.scala]: Removed. I have also added a test in LoggedUserTest to test the roles to permissions mapping.

File: engine/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/ExampleOAuth2ServiceFactory.scala [hunk 4]
```diff
@@ -88,6 +94,15 @@ object ExampleOAuth2ServiceFactory {
       None
     )
 
+  val testRules: List[AuthenticationConfiguration.ConfigRule] = AuthenticationConfiguration.getRules(testConfig.usersFile)
+//  def testRules = List(
```
[the remark this answers is not in the document]
Comment (trombka) [reply engine/security/src/test/scala/pl/touk/nussknacker/ui/security/oauth2/ExampleOAuth2ServiceFactory.scala]: Removed.

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala (hunk 2 above)
Comment (trombka) [reply engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: Right. Garbage. I thought of passing roles along with permissions in LoggedUser but then dropped the idea. Removed.

File: engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala (hunk 1 above)
Comment (trombka) [reply engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: I have moved config reading to the main application class.

Comment (trombka) [reply engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala]: A separate commit so you can see the difference.

**Review by mproch — APPROVED** [review]

**Pull request merged**